client/src/components/checkout-form.js

```diff
@@ -29,22 +29,33 @@ class CheckoutForm extends LitElement {
   constructor() {
     super();
     this.state = {
-      disableSubmit: false,
+      disableSubmit: true,
       openFormErrorDialog: false,
     };
 
     // Bind "this" component to functions
     this.onSubmit = () => {};
     this.submitForm = this.submitForm.bind(this);
+    this.onEmailChange = this.onEmailChange.bind(this);
   }
 
-  toggleFormErrorDialog() {
-    this.state.openFormErrorDialog = !this.state.openFormErrorDialog;
+  firstUpdated() {
+    // Add input change callback for email input
+    const email = this.shadowRoot.getElementById('email');
+    email.addEventListener('change', this.onEmailChange);
+  }
+
+  onEmailChange(event) {
+    const form = new FormData(this.shadowRoot.querySelector('form') || {});
+    const isValid = this.isValidEmail(form.get('email'));
+
+    // Disable submit while form is being sent
+    this.state.disableSubmit = !isValid;
     this.requestUpdate();
   }
 
   isValidEmail(text) {
-    return /[^ @]*@[^ @]*/.test(text);
+    return /^([\w\.\-]+)@([\w\-]+)((\.(\w){2,3})+)$/.test(text);
```

Comment:
suggestion: This regex doesn't validate the full range of email addresses, and while that doesn't matter much for this app, folks may copy this to a production app.

user+label@example.com (+label denied).

Reply:
Agreed, would suggest reverting this change

```suggestion
    return /[^ @]*@[^ @]*/.test(text);
```